newIDE/app/src/BehaviorsEditor/Editors/BehaviorPropertiesEditor.js

```diff
@@ -7,30 +7,78 @@ import propertiesMapToSchema from '../../PropertiesEditor/PropertiesMapToSchema'
 import EmptyMessage from '../../UI/EmptyMessage';
 import { Column } from '../../UI/Grid';
 import { type BehaviorEditorProps } from './BehaviorEditorProps.flow';
+import { ColumnStackLayout } from '../../UI/Layout';
+import Text from '../../UI/Text';
+import { type Schema } from '../../PropertiesEditor';
 
 type Props = BehaviorEditorProps;
 
 export default class BehaviorPropertiesEditor extends React.Component<Props> {
+  _getPropertyGroupNames = (): Array<string> => {
+    const { behavior, behaviorContent } = this.props;
+    const properties = behavior.getProperties(behaviorContent.getContent());
+
+    const groupNames = new Set<string>();
+    const propertyNames = properties.keys();
+    for (let i = 0; i < propertyNames.size(); i++) {
```

Comment:
nitpicking: you can use keys().toJSArray().forEach if you prefer to manipulate a JS array

Reply:
I think something like:

return [...new Set<string>(properties.keys().toJSArray().map(propertyName => property.get(propertyName).getGroup() || "")).sort(...)

can work (just a matter of slightly more functional style)